storage/jellyfish-merkle/src/node_type/mod.rs

```diff
@@ -458,7 +459,8 @@ impl InternalNode {
                         let only_child_version = self
                             .child(only_child_index)
                             .unwrap_or_else(|| {
-                                panic!(
+                                // Should be guaranteed by the self invariants, but these are not easy to express at the moment
+                                assume_unreachable!(
```

Comment:
so wouldn't we want MIRAI to actually check these instead of assuming these are unreachable?

Reply:
assume_unreachable! does a runtime check. It is equivalent to panic a runtime. But at developer time it documents that this should never happen. Panic makes no such statement.